### DISCUSSION TITLE
Handling of queue messages with delivery count exceeding a threshold

### DISCUSSION THREAD

**Discussion by yukscck** [Azure/azure-functions-durable-extension#1924] [Q&A] [answered]
Is there any chance for implementing some sort of poison message handling mechanism in my application, particularly on Azure Queue Storage messages used by Durable Functions?

I am aware that, in applications built on frameworks like the Durable Functions, implementing error handling in the applications mitigates situations of processing messages that results to error outcomes. However, in my application, I have been noticing from occurrences where messages with a high dequeue count in the storage.

In my application, depending on user configurations, each orchestration instance could be executed with very different execution patterns. With the dynamic scenarios my application handles, it is not easy to cover all edges. Hence, I'm looking at a blanket approach, on top of necessary error handlings that are already in place in my application, to mitigate situations of a few anomalous message causing issues to the entire application.

Thanks!

**Comment by cgillum** [comment, marked as answer]
Poison message handling is something we've been wanting to implement as a built-in feature for a while. I'm not sure if that's what you're looking for, but here is an issue tracking this potential feature: https://github.com/Azure/azure-functions-durable-extension/issues/338.

> However, in my application, I have been noticing from occurrences where messages with a high dequeue count in the storage.

If you're running your app in the Azure Functions hosted service and haven't done so already, I recommend taking a look at the "Diagnose and Solve Problems" blade in the Azure portal when navigating to your function app. There is a "Durable Functions" detector which can point out a few potential causes of this problem, like application restarts, internal errors, out-of-date extension versions, etc. See if that can help you identify why you're seeing these increasing dequeue counts.

If you're still unable to identify the source of the problem, consider opening a support ticket with the Azure Functions team. They should be able to help diagnose the problem you're seeing.
